### ISSUE TITLE
Support Puppeteer-Sharp in a Windows Docker

### ISSUE THREAD

**Issue by respel** [kblok/puppeteer-sharp#319]
### Description

I am not able to run Puppeteer Sharp in a Windows docker container on my machine. Though I have tried it on some other machines on which it is working. 

### Complete minimal example reproducing the issue

```csharp
            var options = new LaunchOptions
            {
                Headless = true
            };

            Console.WriteLine("Downloading chromium");
            await Downloader.CreateDefault().DownloadRevisionAsync(Downloader.DefaultRevision);

            Console.WriteLine("Navigating google");
            using (var browser = await Puppeteer.LaunchAsync(options, Downloader.DefaultRevision))
            using (var page = await browser.NewPageAsync())
            {
                await page.GoToAsync("http://www.google.com");

                Console.WriteLine("Generating PDF");
                await page.PdfAsync(Path.Combine(Directory.GetCurrentDirectory(), "google.pdf"));

                Console.WriteLine("Export completed");
                Console.ReadLine();
            }
```

### Expected behavior:

No errors . 

### Actual behavior:

```
Downloading chromium
Navigating google

Unhandled Exception: System.InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at PuppeteerSharp.CDPSession.OnClosed()
   at PuppeteerSharp.Connection.OnClose()
   at PuppeteerSharp.Connection.Dispose()
   at PuppeteerSharp.Browser.<CloseAsync>d__48.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharp.Browser.Dispose()
   at PuppeteerSharpTest.Program.<MainAsync>d__1.MoveNext() in C:\buildir\Program.cs:line 42
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PuppeteerSharpTest.Program.Main(String[] args) in C:\buildir\Program.cs:line 14
```
### Versions

* PuppeteerSharp - v1.0.1

Docker
```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:12:48 2018
 OS/Arch:      windows/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.24)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:21:42 2018
  OS/Arch:      windows/amd64
  Experimental: false
```

Dotnet
```
.NET Command Line Tools (2.1.200)

Product Information:
 Version:            2.1.200
 Commit SHA-1 hash:  2edba8d7f1

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.200\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.7
  Build    : 2d61d0b043915bc948ebf98836fefe9ba942be11
```
### Additional Information

Dockerfile
```dockerfile
FROM microsoft/dotnet:sdk as build
WORKDIR /buildir
COPY . /buildir
RUN dotnet build
RUN dotnet publish -o app


FROM microsoft/windowsservercore
RUN @powershell -NoProfile -ExecutionPolicy unrestricted -Command "iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))" && SET PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin
RUN choco install dotnetcore -y
WORKDIR /app
COPY --from=build /buildir/app /app
ENTRYPOINT ["dotnet", "PuppeteerSharpTest.dll"]
```

**Comment by kblok** [comment]
Running in Docker is not fully supported yet, not because of Puppeteer Sharp, but because some missing dependencies Chromium needs to be able to run on a docker container.
I trying to get a Linux image as part of a side-research.

But your info shows me something we can improve.

**Comment by respel** [comment]
@kblok I was able to make it work on a different machine.

**Comment by kblok** [comment]
@respel using docker?
If so, Could you tell me what OS you have on each machine? and also if it's 32 or 64bits.

**Comment by respel** [comment]
I updated my Windows OS and now it's working on the old machine too.

**Comment by kblok** [comment]
Awesome!

**Comment by Abrissirba** [comment]
@respel  Any guidance on how you managed to get it to work in a windows docker container? I'm getting "Failed to launch chrome" when running in docker.

**Comment by RafaInformatico** [comment]
@Abrissirba we are also trying to run puppeteer sharp in a windows docker container. Just started the process now. Please let us know if you find the correct configuration. We will do the same if we make it work.
We are learning how to use docker containers because we can not run chromedriver nor puppeteer sharp in Azure Web App Services. Our last bullet is browserless.io but it doesn't meet our desired performance requirements.

**Comment by Abrissirba** [comment]
I'm still having issues on windows containers. I switched from using nanoserver to servercore and got passed the "Puppeteer fail to launch" but I'm now stuck on another problem. I can launch a browser and add new pages but I am stuck at goToAsync. If I for exaple go to google.com I get "Navigation failed because browser has disconnected!". I have tried about 5 urls and all but a very simple one, http://vecka.nu, throws the exception. It feels like it is a problem with the server build or that some dependencies are missing but I have no idea how to go forward. 

I tried running the same program in a linux container and it worked perfectly. However, I'm not sure if we can use linux containers so my preferred choice would be to have a working windows container with puppeteer sharp
